diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,11 +81,17 @@ if(${BUILD_FOR_GPU})
     endif()
     message(STATUS "OpenMP_C_FLAGS: ${OpenMP_C_FLAGS}")
     message(STATUS "EXTRA_OpenMP_C_FLAGS: ${EXTRA_OpenMP_C_FLAGS}")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${EXTRA_OpenMP_C_FLAGS}")
+    separate_arguments(
+        EXTRA_OpenMP_C_FLAGS_LIST
+        NATIVE_COMMAND
+        "${EXTRA_OpenMP_C_FLAGS}")
+    if(EXTRA_OpenMP_C_FLAGS_LIST)
+        target_compile_options(muMatScale PRIVATE ${EXTRA_OpenMP_C_FLAGS_LIST})
+        target_link_options(muMatScale PRIVATE ${EXTRA_OpenMP_C_FLAGS_LIST})
+    endif()
     target_compile_definitions(muMatScale PRIVATE GPU_OMP)
     target_compile_definitions(muMatScale PRIVATE GPU_PACK)
     target_compile_definitions(muMatScale PRIVATE GPU_OMP_NUC)
-    target_link_libraries(muMatScale PRIVATE ${EXTRA_OpenMP_C_FLAGS})
     if(NOT ${BUILD_WITH_GPU_AWARE_MPI})
         target_compile_definitions(muMatScale PRIVATE CPU_MPI)
     endif()
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -13,9 +13,10 @@ target_link_libraries(testPacking PRIVATE m)
 if(${BUILD_FOR_GPU})
   target_compile_definitions(testPacking PRIVATE GPU_PACK)
   target_link_libraries(testPacking PRIVATE OpenMP::OpenMP_C)
-  #set_target_properties(testPacking PROPERTIES COMPILE_FLAGS ${EXTRA_OpenMP_C_FLAGS})
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${EXTRA_OpenMP_C_FLAGS}")
-  target_link_libraries(testPacking PRIVATE ${EXTRA_OpenMP_C_FLAGS})
+  if(EXTRA_OpenMP_C_FLAGS_LIST)
+    target_compile_options(testPacking PRIVATE ${EXTRA_OpenMP_C_FLAGS_LIST})
+    target_link_options(testPacking PRIVATE ${EXTRA_OpenMP_C_FLAGS_LIST})
+  endif()
 endif()
 
 add_test(NAME testPacking
