diff --git a/src/grain.c b/src/grain.c
--- a/src/grain.c
+++ b/src/grain.c
@@ -1137,7 +1137,7 @@ cell_nucleation(
                         initnuc = 0;
                 }
 
-                if ((!sb->mold[idx]) && (gr[idx] <= 0) && lyaerno
+                if ((!sb->mold[idx]) && (gr[idx] <= 0) && layerno
                     && initnuc)
                     // Liquid, and a nuc site!
                 {
diff --git a/src/growth.c b/src/growth.c
--- a/src/growth.c
+++ b/src/growth.c
@@ -787,6 +787,8 @@ capture_octahedra_diffuse(
     {
         int idx = diff_id[i];
 
+        if (gr[idx] == ogr[idx])
+            continue;
         dc[idx].x = dc_tmp[i].x;
         dc[idx].y = dc_tmp[i].y;
         dc[idx].z = dc_tmp[i].z;
